koku/masu/database/trino_sql/reporting_ocpazurecostlineitem_daily_summary.sql

```diff
@@ -599,6 +600,113 @@ JOIN cte_rankings as r
+    persistentvolume,
+    storageclass,
+    pod_labels,
+    resource_id,
+    usage_start,
+    usage_end,
+    cost_entry_bill_id,
+    subscription_guid,
+    subscription_name,
+    instance_type,
+    service_name,
+    resource_location,
+    usage_quantity,
+    unit_of_measure,
+    currency,
+    pretax_cost,
+    markup_cost,
+    project_markup_cost,
+    pod_cost,
+    tags,
+    cost_category_id,
+    source_uuid
+)
+WITH cte_filtered_data as (
+    SELECT
+        coalesce(ocp_filtered.date, ocp_filtered.usagedatetime) AS usage_date,
+        coalesce(nullif(ocp_filtered.resourceid, ''), ocp_filtered.instanceid) as resource_id
+    FROM azure_openshift_daily as ocp_filtered
+    WHERE ocp_filtered.source = {{azure_source_uuid}}
+        AND ocp_filtered.year = {{year}}
+        AND ocp_filtered.month = {{month}}
+        AND coalesce(ocp_filtered.date, ocp_filtered.usagedatetime) >= TIMESTAMP '{{start_date | sqlsafe}}'
+        AND coalesce(ocp_filtered.date, ocp_filtered.usagedatetime) < date_add('day', 1, TIMESTAMP '{{end_date | sqlsafe}}')
+        AND coalesce(nullif(ocp_filtered.servicename, ''), ocp_filtered.metercategory) LIKE '%Storage%'
+        AND coalesce(nullif(ocp_filtered.resourceid, ''), ocp_filtered.instanceid) LIKE '%%Microsoft.Compute/disks/%%'
+        AND ocp_filtered.resource_id_matched = TRUE
+)
+SELECT uuid(),
+    {{report_period_id | sqlsafe}} as report_period_id,
+    cast(NULL as varchar) as cluster_id,
+    cast(NULL as varchar) as cluster_alias,
+    'Storage' as data_source,
+    'Storage unattributed' as namespace,
+    cast(NULL as varchar) as node,
+    cast(NULL as varchar) as persistentvolumeclaim,
+    cast(NULL as varchar) as persistentvolume,
+    cast(NULL as varchar) as storageclass,
+    cast(map(array[], array[]) as json) as pod_labels,
+    split_part(coalesce(nullif(azure.resourceid, ''), azure.instanceid), '/', 9) as resource_id,
+    date(coalesce(azure.date, azure.usagedatetime)) as usage_start,
+    date(coalesce(azure.date, azure.usagedatetime)) as usage_end,
+    {{bill_id | sqlsafe}} as cost_entry_bill_id,
+    coalesce(nullif(azure.subscriptionid, ''), azure.subscriptionguid) as subscription_guid,
+    max(azure.subscriptionname) as subscription_name,
+    max(json_extract_scalar(json_parse(azure.additionalinfo), '$.ServiceType')) as instance_type,
+    coalesce(nullif(azure.servicename, ''), azure.metercategory) as service_name,
+    azure.resourcelocation as resource_location,
+    sum(cast(coalesce(nullif(azure.quantity, 0), azure.usagequantity) as decimal(24, 9))) as usage_quantity,
+    max(CASE
+        WHEN split_part(azure.unitofmeasure, ' ', 2) = 'Hours'
+            THEN  'Hrs'
+        WHEN split_part(azure.unitofmeasure, ' ', 2) = 'GB/Month'
+            THEN  'GB-Mo'
+        WHEN split_part(azure.unitofmeasure, ' ', 2) != ''
+            THEN  split_part(azure.unitofmeasure, ' ', 2)
+        ELSE azure.unitofmeasure
+    END) as unit_of_measure,
```

Comment:
Does storage data have all of these different units and do we need to do any unit conversion on the quantity?